web/packages/teleport/src/DesktopSession/DesktopSession.tsx

```diff
@@ -186,9 +186,9 @@ const MfaDialog = ({ mfa }: { mfa: MfaState }) => {
     <AuthnDialog
       mfa={mfa}
       onCancel={() => {
-        mfa.setErrorText(
-          'This session requires multi factor authentication to continue. Please hit "Retry" and follow the prompts given by your browser to complete authentication.'
-        );
+        mfa.submitAttempt.status = 'failed';
+        mfa.submitAttempt.statusText =
```

Comment:
This may result in a subtle bug. First of all, we are modifying something that looks like an internal state of some other process. We should keep reporting status a function exported on the `MfaState` type to hermetize data access — just instead of just setting error text, it should change the entire attempt status. Second, this modification will mean that we change parameters that may influence rendering, `useCallback`, etc., without creating a new object and without going through `setAttempt` function returned by `useAttempt`. This may mean breaking the React change detection mechanism.